src/components/contextual/pages/pool/PoolChart.vue

```diff
@@ -21,198 +23,309 @@ type Props = {
   pool: Pool;
 };
 
+enum PoolChartTab {
+  VOLUME = 'volume',
+  TVL = 'tvl',
+  FEES = 'fees'
+}
+
 /**
  * PROPS
  */
 const props = withDefaults(defineProps<Props>(), {
   loading: false
 });
 
-/**Z
- * STATE
- */
-const MIN_CHART_VALUES = 7;
-
 /**
  * COMPOSABLES
  */
 const store = useStore();
 const { t } = useI18n();
 const tailwind = useTailwind();
+const { fNum2 } = useNumbers();
+const { isMobile } = useBreakpoints();
 const { darkMode } = useDarkMode();
 
 /**
- * COMPUTED
+ * STATE
  */
-const hodlColor = computed(() =>
-  darkMode.value
-    ? tailwind.theme.colors.gray['600']
-    : tailwind.theme.colors.black
-);
-
-const chartColors = computed(() => [
-  tailwind.theme.colors.green['400'],
-  hodlColor.value
-]);
+const MIN_CHART_VALUES = 2;
+
+const tabs = [
+  {
+    value: PoolChartTab.VOLUME,
+    label: t('poolChart.tabs.volume')
+  },
+  {
+    value: PoolChartTab.TVL,
+    label: t('poolChart.tabs.tvl')
+  },
+  {
+    value: PoolChartTab.FEES,
+    label: t('poolChart.tabs.fees')
+  }
+];
+const activeTab = ref(tabs[0].value);
 
-const supportsPoolLiquidity = computed(() =>
-  isStablePhantom(props.pool.poolType)
-);
+const currentChartValue = ref('');
+const currentChartDate = ref('');
+const isFocusedOnChart = ref(false);
 
+/**
+ * COMPUTED
+ */
 const appLoading = computed(() => store.state.app.loading);
 
-const history = computed(() => {
-  if (!props.historicalPrices) return [];
+const snapshotValues = computed(() => Object.values(props.snapshots || []));
 
-  const pricesTimestamps = Object.keys(props.historicalPrices);
-  const snapshotsTimestamps = Object.keys(props.snapshots);
+const periodOptions = computed(() => {
+  const maxPeriodLengh = snapshotValues.value.length;
+  const arr = [{ text: t('poolChart.period.all'), value: maxPeriodLengh }];
 
-  if (snapshotsTimestamps.length === 0) {
-    return [];
+  if (maxPeriodLengh > 365) {
```

Comment:
I think we should show all periods no longer how old the pool is and just only show the data we have. Otherwise it may seem the menu is broken / missing options.

Reply:
fixed, now all periods are shown